### DISCUSSION TITLE
How can i get user's ip when using frp

### DISCUSSION THREAD

**Discussion by ravizhan** [Pryaxis/TShock#2551] [Q&A]
I am using **frp** in order to let other players join in, however it seems that i can't find user's ip, all i get is the **frp server ip**.
I think that may be **Proxy protocol** can make it. But i can't find anything about it in [USER DOCS](https://tshock.readme.io/docs), i'm wondering if it is supported by TShock.TKS

**Comment by bartico6** [comment]
Proxy protocol is not supported by tShock. You will have to develop your own plugin to synchronise with frp.
